docs/admin/architectures/index.md

```diff
@@ -171,3 +171,167 @@ Database:
+interruptions for user connections, see [Autoscaling](../scale.md#autoscaling)
+for more details.
+
+### Control plane: provisionerd
+
+Each provisioner can run a single concurrent workspace build. For example,
+running 10 provisioner containers will allow 10 users to start workspaces at the
+same time.
+
+By default, the Coder server runs built-in provisioner daemons, but the
+_Enterprise_ Coder release allows for running external provisioners to separate
+the load caused by workspace provisioning on the `coderd` nodes.
+
+#### Scaling formula
+
+When determining scaling requirements, consider the following factors:
+
+- `1 vCPU x 1 GB memory x 2 concurrent workspace build`: A formula to determine
+  resource allocation based on the number of concurrent workspace builds, and
+  standard complexity of a Terraform template. _Rule of thumb_: the more
+  provisioners are free/available, the more concurrent workspace builds can be
+  performed.
+
+**Node Autoscaling**
+
+Autoscaling provisioners is not an easy problem to solve unless it can be
+predicted when a number of concurrent workspace builds increases.
+
+We recommend disabling autoscaling and adjusting the number of provisioners to
+developer needs based on the workspace build queuing time.
+
+### Data plane: Workspaces
+
+To determine workspace resource limits and keep the best developer experience
+for workspace users, administrators must be aware of a few assumptions.
+
+- Workspace pods run on the same Kubernetes cluster, but possible in a different
+  namespace or a node pool.
+- Workspace limits (per workspace user):
+  - Evaluate the workspace utilization pattern. For instance, a regular web
+    development does not require high CPU capacity all the time, but only during
+    project builds or load tests.
+  - Evaluate minimal limits for single workspace. Include in the calculation
+    requirements for Coder agent running in an idle workspace - 0.1 vCPU and 256
+    MB. For instance, developers can choose between 0.5-8 vCPUs, and 1-16 GB
+    memory.
+
+#### Scaling formula
+
+When determining scaling requirements, consider the following factors:
+
+- `1 vCPU x 2 GB memory x 1 workspace`: A formula to determine resource
+  allocation based on the minimal requirements for an idle workspace with a
+  running Coder agent and occasional CPU and memory bursts for building
+  projects.
+
+**Node Autoscaling**
+
+Workspace nodes can be set to operate in autoscaling mode to mitigate the risk
+of prolonged high resource utilization.
+
+One approach is to scale up workspace nodes when total CPU usage or memory
+consumption reaches 80%. Another option is to scale based on metrics such as the
+number of workspaces or active users. It's important to note that as new users
+onboard, the autoscaling configuration should account for ongoing workspaces.
+
+Scaling down workspace nodes to zero is not recommended, as it will result in
+longer wait times for workspace provisioning by users.
```

Comment:
```suggestion
Scaling down workspace nodes to zero is not recommended, as it will result in
longer wait times for workspace provisioning by users. However, this may be
necessary for workspaces with special resource requirements (e.g. GPUs) that incur significant cost overheads.
```